ls1intum/Artemis
# `Communication`: Improve usability of conversations user interface

### Checklist
#### General
#### Client

### Motivation and Context + Description

Various small UI improvements requested by @krusche :

- Moved the filter icon to the right side of the input. The filter icon blue when active and it shows the number of conversations.
![image](https://user-images.githubusercontent.com/29718932/215119837-dc458203-cc6c-45af-b02d-b0c1cb6cbe5f.png)
**Old:**
![image](https://user-images.githubusercontent.com/29718932/215124040-35d1d48c-cc9e-48eb-9426-820247672abf.png)

- Always now show the "plus" icon in the conversation sidebar. Also remove the blue color for active conversations and instead use a darker color to signal something is highlighted or active. Also changed the look of the buttons.
![image](https://user-images.githubusercontent.com/29718932/215120254-632f57ae-8312-47c6-a554-4d6553868fe7.png)
![image](https://user-images.githubusercontent.com/29718932/215120268-c2e22b71-cc0e-4593-8641-e3e76452a0d1.png)
![image](https://user-images.githubusercontent.com/29718932/215120283-e03792a4-7955-491c-abd4-25b1f3ea4192.png)
**Old:**
![image](https://user-images.githubusercontent.com/29718932/215124697-539dcd2d-1976-43ee-a96b-7e342addf287.png)
![image](https://user-images.githubusercontent.com/29718932/215124189-dbe115ca-def8-484d-bd8a-85c25335ab76.png)

- Do not show login name any more in the conversation sidebar.
![image](https://user-images.githubusercontent.com/29718932/215120546-c69375cb-e14a-46c7-aa45-5b83e2e9620e.png)
**Old:**
![image](https://user-images.githubusercontent.com/29718932/215124343-1b8dcaaa-2546-4af9-8018-37ddaa5ffdd1.png)

- Remove the chevron icon in the conversation header.
![image](https://user-images.githubusercontent.com/29718932/215120558-59c35bc5-9892-4c4e-b9c5-01084ef0a87d.png)
**Old:**
![image](https://user-images.githubusercontent.com/29718932/215124402-b16d2639-5137-46af-9606-260ec75808a9.png)

- Do not show id any more in info tab of conversation
![image](https://user-images.githubusercontent.com/29718932/215120648-38e96c16-112a-44f0-a2e7-be9279e4938a.png)
**Old:**
![image](https://user-images.githubusercontent.com/29718932/215124447-ac90055d-5a31-4381-9e87-ddc9b205fb1d.png)

- Change button to outline in channel creation form to make it more clear which option is selected.
![image](https://user-images.githubusercontent.com/29718932/215120811-c209f22c-3fc3-4d8c-97d6-75196a3ae140.png)
**Old:**
![image](https://user-images.githubusercontent.com/29718932/215124488-68f5e92c-575e-49d6-b15d-f4287ea61918.png)

- Changing the favorite status or the hidden status of a conversation does not cause a reload any more, meaning the loading indicator does not show up.

- Removed the dividers in the conversation sidebar and instead use margin for separation:
![image](https://user-images.githubusercontent.com/29718932/215274253-ff2aeb74-0bae-4798-8124-e0d02b988383.png)

**Old:**
![image](https://user-images.githubusercontent.com/29718932/215273861-77b2c744-9f5d-460d-9876-4f6124c03d09.png)

- Removed the collapse all and expand all buttons as it is rarely used feature and clutters the UI. 
![image](https://user-images.githubusercontent.com/29718932/215274539-9ba24987-00b6-4db6-9f26-aa3247c21e06.png)

- Make the border of the search field flush with the sidebar

![image](https://user-images.githubusercontent.com/29718932/215275229-3a4c42c2-2f39-47a3-852d-684b48871932.png)

**Old:**
![image](https://user-images.githubusercontent.com/29718932/215274900-cae59b4f-3f4f-4485-a4b8-2397014f2609.png)

- Decreased size of chevrons:

![image](https://user-images.githubusercontent.com/29718932/215275397-abcefd6b-ea36-4fb4-858f-4704c87f654b.png)

**Old:**
![image](https://user-images.githubusercontent.com/29718932/215275316-544c001c-744a-4cb1-8df0-f2c4e7bc804e.png)


- Only keep the top border of the message input to make design cleaner

![image](https://user-images.githubusercontent.com/29718932/215275774-aeb4c1ef-d3ab-45d0-913d-28394f906a8d.png)

**Old: **

![image](https://user-images.githubusercontent.com/29718932/215275788-6b502043-3028-43ed-915c-b54d78258b62.png)


- Improved look and usability of search for messages input


https://user-images.githubusercontent.com/29718932/215533529-d1972c7c-9f27-4cc3-89ce-c2be00cd6fa2.mp4

**Old: **

![image](https://user-images.githubusercontent.com/29718932/215533777-6b83d3e9-e4c3-4bea-b1eb-080ba8cff347.png)

- Make text input close to header when few messages are posted:
 
![image](https://user-images.githubusercontent.com/29718932/215536820-4bb6abab-5948-4d32-b45e-10625b431704.png)

**Old: **

![image](https://user-images.githubusercontent.com/29718932/215537371-5feacd1c-fb94-4abf-9041-35fd3b3b48b6.png)



### Steps for Testing
- Send each other messages in channels, group chats and direct messages and check that everything works
- Please also checks threads are working as I touched the logic here




### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
- No Changes
